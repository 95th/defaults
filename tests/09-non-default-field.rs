use defaults::Defaults;

struct NonDefault;

#[derive(Defaults)]
struct Foo {
    x: usize,
    non_default: NonDefault,
}

fn main() {}
