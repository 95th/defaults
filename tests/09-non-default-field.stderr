error[E0277]: the trait bound `NonDefault: Default` is not satisfied
 --> tests/09-non-default-field.rs:8:5
  |
8 |     non_default: NonDefault,
  |     ^^^^^^^^^^^ the trait `Default` is not implemented for `NonDefault`
  |
help: consider annotating `NonDefault` with `#[derive(Default)]`
  |
3 + #[derive(Default)]
4 | struct NonDefault;
  |
